co-authors-plus.php

```diff
@@ -600,22 +600,23 @@ public function update_author_term_post_count( $term ) {
 		$query = "SELECT COUNT({$wpdb->posts}.ID) FROM {$wpdb->posts}";
 
 		$query .= " LEFT JOIN {$wpdb->term_relationships} ON ({$wpdb->posts}.ID = {$wpdb->term_relationships}.object_id)";
-		$query .= " LEFT JOIN {$wpdb->term_taxonomy} ON ( {$wpdb->term_relationships}.term_taxonomy_id = {$wpdb->term_taxonomy}.term_taxonomy_id )";
-
-		$having_terms_and_authors = $having_terms = $wpdb->prepare( "{$wpdb->term_taxonomy}.term_id = %d", $term->term_id );
-		if ( 'wpuser' == $coauthor->type ) {
-			$having_terms_and_authors .= $wpdb->prepare( " OR {$wpdb->posts}.post_author = %d", $coauthor->ID );
-		}
-
+		$query .= " LEFT JOIN {$wpdb->terms} ON ({$wpdb->term_relationships}.term_taxonomy_id = {$wpdb->terms}.term_id)";
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users
+		$query .= " LEFT JOIN {$wpdb->users} ON ({$wpdb->posts}.post_author = {$wpdb->users}.ID)";
+	
 		$post_types = apply_filters( 'coauthors_count_published_post_types', array( 'post' ) );
 		$post_types = array_map( 'sanitize_key', $post_types );
 		$post_types = "'" . implode( "','", $post_types ) . "'";
 
-		$query .= " WHERE ({$having_terms_and_authors}) AND {$wpdb->posts}.post_type IN ({$post_types}) AND {$wpdb->posts}.post_status = 'publish'";
-
-		$query .= $wpdb->prepare( " GROUP BY {$wpdb->posts}.ID HAVING MAX( IF ( {$wpdb->term_taxonomy}.taxonomy = '%s', IF ( {$having_terms},2,1 ),0 ) ) <> 1 ", $this->coauthor_taxonomy ); //phpcs:ignore
+		$query .= " WHERE {$wpdb->posts}.post_type IN ({$post_types})";
+		$query .= " AND {$wpdb->posts}.post_status = 'publish'";
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users
+		$query .= " AND ( {$wpdb->users}.user_login = %s OR {$wpdb->terms}.name = %s )";
+		
+		$query .= " GROUP BY {$wpdb->posts}.ID";
 
-		$count = $wpdb->query( $query ); // phpcs:ignore
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$count = $wpdb->query( $wpdb->prepare( $query, $coauthor->linked_account, $coauthor->user_login ) ); 
```

Comment:
This assumes that there is a linked account, but what if there's no linked account?

Reply:
I should have mentioned that I already checked what happens if there's no linked account. 🤭

When there's no account linked, the variable `$coauthor->linked_account` returns an empty string. Testing this with a post that had been created as `admin` with no linked account, the query looks as follows:

```sql
SELECT COUNT(wp_posts.ID)
FROM wp_posts
LEFT JOIN wp_term_relationships ON (wp_posts.ID = wp_term_relationships.object_id)
LEFT JOIN wp_terms ON (wp_term_relationships.term_taxonomy_id = wp_terms.term_id)
LEFT JOIN wp_users ON (wp_posts.post_author = wp_users.ID)
WHERE wp_posts.post_type IN ('post')
AND wp_posts.post_status = 'publish'
AND ( wp_users.user_login = '' OR wp_terms.name = 'admin' )
GROUP BY wp_posts.ID
```

I thought about checking if `$coauthor->linked_account` actually has a value, but given that it returns an empty string which then leads to the correct post count, I decided against it. 

Would you advise practising defensive coding and checking if `$coauthor->linked_account` has a value other than an empty string?